### ISSUE TITLE
ExpressionAttributeValues seem to not support props with underscores

### ISSUE THREAD

**Issue by danikenan** [sam-goodwin/typesafe-dynamodb#58]
Seems like the string pattern used does not take into account underscores in property names.

![Screenshot 2024-10-08 at 9 51 51](https://github.com/user-attachments/assets/68fa7915-1a6b-4042-987e-4507f962ca42)

```
interface MyItem {
  key: string;
  noUnderscore: string;
  has_underscore: string;
}

const MyItemUpdateCommand = TypeSafeUpdateItemCommand<MyItem, 'key', undefined>();

async function updateIt1(key: string, value: string) {
  return dynamo.send(
    new MyItemUpdateCommand({
      TableName: 'MyItemTable',
      Key: { key: { S: key } },
      UpdateExpression: `set noUnderscore = :noUnderscore`,
      ExpressionAttributeValues: { ':noUnderscore': { S: value } },
    })
  );
}

async function updateIt2(key: string, value: string) {
  return dynamo.send(
    new MyItemUpdateCommand({
      TableName: 'MyItemTable',
      Key: { key: { S: key } },
      UpdateExpression: `set has_underscore = :has_underscore`,
      ExpressionAttributeValues: { ':has_underscore': { S: value } },
    })
  );
}
```

**Comment by sam-goodwin** [comment]
Thanks for reporting, it does indeed look like there's a problem in how identifiers were being parsed. Fixed in: https://github.com/sam-goodwin/typesafe-dynamodb/pull/59

**Comment by sam-goodwin** [comment]
Should be fixed in 0.8.0. Let me know if it works for you!

**Comment by sam-goodwin** [comment]
Should be fixed in v0.8.1, LMK if it works!
